### ISSUE TITLE
'fig run' output discarded or truncated sometimes

### ISSUE THREAD

**Issue by umrashrf** [docker/compose#177]
In a running container when I run ```fig run web ls /``` the output is usually discarded and sometimes truncated and is rarely shown in full.

I found a workaround and this works ```fig run web sh -c 'sleep 0.1; ls /'```.

**Comment by sherter** [comment]
This is still an issue.
Consider this yml file:
```
echo:
  image: ubuntu
  command: echo "foo"
```


The output of `docker-compose run echo` is unpredictable. On my machine it outputs `foo` sometimes 2 times, sometimes only once (with newlines in between?!). That's really strange!

The problem doesn't occur if I change my yml file to 
```
echo:
  image: ubuntu
  command: sh -c 'sleep 0.1; echo "foo"'
```

**Comment by aanand** [comment]
Duplicate of #658, will hopefully be fixed by https://github.com/d11wtq/dockerpty/pull/36.
